dask/dataframe/tests/test_utils_dataframe.py

```diff
@@ -233,7 +233,7 @@ def test_meta_duplicated():
 
     exp = pd.DataFrame(
         [["foo", "foo", "foo"], ["foo", "foo", "foo"]],
-        index=["a", "b"],
+        index=meta_nonempty(df.index),
```

Comment:
This looks magical, what does it do exactly?

Reply:
Dask DataFrame tracks column / dtype information by keeping track of an empty `._meta` object (a pandas Series or DataFrame). This mostly works, but some operations require a non-empty meta object which is what `meta_nonempty` is used for generating. https://docs.dask.org/en/stable/dataframe-design.html#metadata has some details. Hopefully that's helpful.